lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,180 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+sinatra_vs = Gem::Version.new(Sinatra::VERSION)
+sinatra_min_vs = Gem::Version.new('1.4.0')
+if sinatra_vs < sinatra_min_vs
+  raise "sinatra version #{sinatra_vs} is not supported yet " \
```

Comment:
I thought a long time about this one. You cannot easily return to stop the evaluation of the tracer module (actually you can since very recently: https://bugs.ruby-lang.org/issues/4840), so you cannot just stop evaluating the module if Sinatra is too old. So it will fail, and the user will have to remove the call to `require 'ddtrace/contrib/sinatra/tracer``, which is fair since it's not compatible. I cannot find any advantage in allowing people to require a module that does not do anything.